packages/framework/fluid-framework/src/index.ts

```diff
@@ -22,85 +22,75 @@ export type {
 	MemberChangedListener,
 	SharedObjectClass,
 } from "@fluidframework/fluid-static";
+export type { ISharedMap, ISharedMapEvents, IValueChanged } from "@fluidframework/map";
+export { SharedMap } from "@fluidframework/map";
+
 export type {
-	ICreateInfo,
-	IDirectory,
-	IDirectoryClearOperation,
-	IDirectoryCreateSubDirectoryOperation,
-	IDirectoryDataObject,
-	IDirectoryDeleteOperation,
-	IDirectoryDeleteSubDirectoryOperation,
-	IDirectoryEvents,
-	IDirectoryKeyOperation,
-	IDirectoryNewStorageFormat,
-	IDirectoryOperation,
-	IDirectorySetOperation,
-	IDirectoryStorageOperation,
-	IDirectorySubDirectoryOperation,
-	IDirectoryValueChanged,
-	ILocalValue,
-	ISerializableValue,
-	ISerializedValue,
-	ISharedDirectory,
-	ISharedDirectoryEvents,
-	ISharedMap,
-	ISharedMapEvents,
-	IValueChanged,
-} from "@fluidframework/map";
-export {
-	DirectoryFactory,
-	LocalValueMaker,
-	MapFactory,
-	SharedDirectory,
-	SharedMap,
-} from "@fluidframework/map";
-export type {
-	DeserializeCallback,
-	IInterval,
-	IIntervalCollectionEvent,
-	IIntervalHelpers,
-	IJSONRunSegment,
-	IMapMessageLocalMetadata,
-	IIntervalCollection,
-	IntervalLocator,
-	ISequenceDeltaRange,
-	ISerializableInterval,
-	ISerializedInterval,
-	ISharedIntervalCollection,
-	ISharedSegmentSequenceEvents,
-	ISharedString,
-	IValueOpEmitter,
-	SerializedIntervalDelta,
-	SharedStringSegment,
-} from "@fluidframework/sequence";
+	AllowedTypes,
+	ApplyKind,
+	ArrayToUnion,
+	CheckoutEvents,
+	Events,
+	ExtractItemType,
+	FlexList,
+	FlexListToUnion,
+	IDisposable,
+	ImplicitAllowedTypes,
+	ImplicitFieldSchema,
+	InsertableObjectFromSchemaRecord,
+	InsertableTreeFieldFromImplicitField,
+	InsertableTreeNodeFromImplicitAllowedTypes,
+	InsertableTypedNode,
+	IsEvent,
+	ISubscribable,
+	ITree,
+	LazyItem,
+	MakeNominal,
+	NodeBuilderData,
+	NodeFromSchema,
+	ObjectFromSchemaRecord,
+	RestrictiveReadonlyRecord,
+	Revertible,
+	TreeApi,
+	TreeArrayNodeBase,
+	TreeFieldFromImplicitField,
+	TreeLeafValue,
+	TreeMapNode,
+	TreeMapNodeBase,
+	TreeNodeEvents,
+	TreeNodeFromImplicitAllowedTypes,
+	TreeNodeSchema,
+	TreeNodeSchemaClass,
+	TreeNodeSchemaCore,
+	TreeNodeSchemaNonClass,
+	TreeView,
+	Unhydrated,
+	WithType,
+} from "@fluidframework/tree";
 export {
-	getTextAndMarkers,
-	Interval,
-	intervalLocatorFromEndpoint,
-	IntervalType,
-	SequenceDeltaEvent,
-	SequenceEvent,
-	SequenceInterval,
-	SequenceMaintenanceEvent,
-	SharedIntervalCollection,
-	SharedIntervalCollectionFactory,
-	SharedSegmentSequence,
-	SharedSequence,
-	SharedString,
-	SharedStringFactory,
-	SubSequence,
-} from "@fluidframework/sequence";
-
-// The tree package manages its own API surface.
-// eslint-disable-next-line no-restricted-syntax
-export * from "@fluidframework/tree";
+	create,
+	DiscardResult,
+	disposeSymbol,
+	FieldKind,
+	FieldSchema,
+	IterableTreeListContent,
+	NodeKind,
+	RevertibleKind,
+	RevertResult,
+	SchemaFactory,
+	SharedTree,
+	Tree,
+	TreeArrayNode,
+	TreeConfiguration,
+	TreeNode,
+	TreeStatus,
+	type,
```

Comment:
At this level this name seems a little too generic. We might want to pick a more specific name like treeNodeTypeSymbol.

Reply:
Same goes for `WithType` Maybe TreeNodeWithType.